db/write_thread.h

```diff
@@ -25,9 +26,29 @@
 #include "rocksdb/types.h"
 #include "rocksdb/write_batch.h"
 #include "util/autovector.h"
-#include "util/safe_queue.h"
 
 namespace rocksdb {
+struct CommitRequest {
+  uint64_t commit_lsn;
+  // We use finished_write to check whether this writer has applied to memtable.
+  std::atomic<bool> finished_write;
+  std::atomic<bool> committed;
+  CommitRequest() : commit_lsn(0), finished_write(false), committed(false) {}
+};
+
+class RequestQueue {
+ public:
+  RequestQueue();
+  ~RequestQueue();
+  void JoinCommitRequest(CommitRequest* req);
+  void CommitSequenceAwait(CommitRequest* req,
+                           std::atomic<uint64_t>* commit_sequence);
```

Comment:
How about accepting `&VersionSet` instead of this

Reply:
write thread is a light structure. VersionSet is too heavy...

Reply:
it's just a reference, and use `versions->SetLastSequence`, it's more comprehensible